cpp/turbodbc_arrow/Library/src/arrow_result_set.cpp

```diff
@@ -0,0 +1,252 @@
+}
+
+Status append_to_double_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t* valid_bytes) {
+  auto typed_builder = static_cast<DoubleBuilder*>(builder.get());
+  auto data_ptr = reinterpret_cast<const double*>(input_buffer.data_pointer());
+  return typed_builder->Append(data_ptr, rows_in_batch, valid_bytes);
+}
+
+Status append_to_int_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t* valid_bytes) {
+  auto typed_builder = static_cast<Int64Builder*>(builder.get());
+  auto data_ptr = reinterpret_cast<const int64_t*>(input_buffer.data_pointer());
+  return typed_builder->Append(data_ptr, rows_in_batch, valid_bytes);
+}
+
+Status append_to_bool_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t* valid_bytes) {
+  auto typed_builder = static_cast<BooleanBuilder*>(builder.get());
+  auto data_ptr = reinterpret_cast<const uint8_t*>(input_buffer.data_pointer());
+  return typed_builder->Append(data_ptr, rows_in_batch, valid_bytes);
+}
+
+Status append_to_timestamp_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t*) {
+  auto typed_builder = static_cast<TimestampBuilder*>(builder.get());
+  for (std::size_t j = 0; j < rows_in_batch; ++j) {
+    auto element = input_buffer[j];
+  	if (element.indicator == SQL_NULL_DATA) {
+      ARROW_RETURN_NOT_OK(typed_builder->AppendNull());
+  	} else {
+  		ARROW_RETURN_NOT_OK(typed_builder->Append(timestamp_to_microseconds(element.data_pointer)));
+  	}
+  }
+  return Status::OK();
+}
+
+Status append_to_date_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t*) {
+  auto typed_builder = static_cast<Date32Builder*>(builder.get());
+  for (std::size_t j = 0; j < rows_in_batch; ++j) {
+    auto element = input_buffer[j];
+  	if (element.indicator == SQL_NULL_DATA) {
+      ARROW_RETURN_NOT_OK(typed_builder->AppendNull());
+  	} else {
+  		ARROW_RETURN_NOT_OK(typed_builder->Append(date_to_days(element.data_pointer)));
+  	}
+  }
+  return Status::OK();
+}
+
+Status append_to_string_builder(size_t rows_in_batch, const std::unique_ptr<ArrayBuilder>& builder, const cpp_odbc::multi_value_buffer& input_buffer, uint8_t*) {
+  auto typed_builder = static_cast<StringBuilder*>(builder.get());
+  for (std::size_t j = 0; j != rows_in_batch; ++j) {
+  	auto const element = input_buffer[j];
+  	if (element.indicator == SQL_NULL_DATA) {
+      ARROW_RETURN_NOT_OK(typed_builder->AppendNull());
+  	} else {
+      ARROW_RETURN_NOT_OK(typed_builder->Append(element.data_pointer, element.indicator));
+  	}
+  }
+  return Status::OK();
+}
+
+Status arrow_result_set::fetch_all_native(std::shared_ptr<arrow::Table>* out)
```

Comment:
Using these return codes comes from the Google C++ Styleguide https://google.github.io/styleguide/cppguide.html#Exceptions that is used for Apache Arrow code. As the `turbodbc_arrow` C++ is mostly Arrow code, I have sticked to it here. Thus also the pointer to the `shared_ptr` as it's an output parameter.